tests/components/systemmonitor/test_sensor.py

```diff
@@ -0,0 +1,267 @@
+"""Test System Monitor sensor."""
+from datetime import timedelta
+import socket
+from unittest.mock import Mock, patch
+
+from freezegun.api import FrozenDateTimeFactory
+from psutil._common import shwtemp, snetio, snicaddr
+from psutil._pslinux import svmem
+import pytest
+
+from homeassistant.components.systemmonitor.sensor import get_cpu_icon
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ON, STATE_UNAVAILABLE, STATE_UNKNOWN
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from .conftest import MockProcess
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+
+
+async def test_sensor(
+    hass: HomeAssistant, mock_added_config_entry: ConfigEntry
+) -> None:
+    """Test the sensor."""
+    memory_sensor = hass.states.get("sensor.system_monitor_memory_free")
+    assert memory_sensor is not None
+    assert memory_sensor.state == "40.0"
+
+    process_sensor = hass.states.get("sensor.system_monitor_process_python3")
+    assert process_sensor is not None
+    assert process_sensor.state == STATE_ON
+
+
+async def test_sensor_icon_32bit() -> None:
+    """Test the sensor icon for 32bit system."""
+
+    with patch("sys.maxsize", 2**32):
+        assert get_cpu_icon() == "mdi:cpu-32-bit"
+    with patch("sys.maxsize", 2**64):
+        assert get_cpu_icon() == "mdi:cpu-64-bit"
+
```

Comment:
Ideally we would patch `sys.maxsize` and write state to verify the icon instead of testing this directly (and i there is a cache it would be ok to clear the cache with `.clear_cache` https://docs.python.org/3/library/functools.html#functools.lru_cache

Reply:
The problem is that the icon is evaluated when the file is loaded so before the test even starts.
No idea how to patch `sys.maxsize` in such case and I don't think clearing a cache helps as the icon has already been set in the entity descriptions then.

Reply:
I can push a suggestion if you like

Reply:
```
diff --git a/tests/components/systemmonitor/test_sensor.py b/tests/components/systemmonitor/test_sensor.py
index 25b6fd901fe..85ea38ddb70 100644
--- a/tests/components/systemmonitor/test_sensor.py
+++ b/tests/components/systemmonitor/test_sensor.py
@@ -5,7 +5,7 @@ from unittest.mock import Mock, patch
 
 from freezegun.api import FrozenDateTimeFactory
 from psutil._common import shwtemp, snetio, snicaddr
-from psutil._pslinux import svmem
+
 import pytest
 from syrupy.assertion import SnapshotAssertion
 
@@ -16,7 +16,7 @@ from homeassistant.core import HomeAssistant
 from homeassistant.helpers import entity_registry as er
 from homeassistant.setup import async_setup_component
 
-from .conftest import MockProcess
+from .conftest import MockProcess, MockSvMem
 
 from tests.common import MockConfigEntry, async_fire_time_changed
 
@@ -69,9 +69,10 @@ async def test_sensor_not_loading_veth_networks(
 
 async def test_sensor_icon() -> None:
     """Test the sensor icon for 32bit/64bit system."""
-
+    get_cpu_icon.cache_clear()
     with patch("sys.maxsize", 2**32):
         assert get_cpu_icon() == "mdi:cpu-32-bit"
+    get_cpu_icon.cache_clear()
     with patch("sys.maxsize", 2**64):
         assert get_cpu_icon() == "mdi:cpu-64-bit"
 
```

But better to patch `sys.maxsize` and clear the cache, move time forward, and wait for the update and verify the icon

Reply:
```python
    with patch("sys.maxsize", 2**32):
        get_cpu_icon.cache_clear()
        mock_config_entry.add_to_hass(hass)
        await hass.config_entries.async_setup(mock_config_entry.entry_id)
        await hass.async_block_till_done()

    process_sensor = hass.states.get("sensor.system_monitor_process_python3")
    assert process_sensor is not None
    assert process_sensor.attributes["icon"] == "mdi:cpu-32-bit"
```
Not helping as it still comes out as 64-bit given my system.
entity descriptions are also a constant so not sure how the cache would help here. I only see it would work by moving it down to the entity but I would prefer to skip that.

Reply:
Feel free to push a change if you find a workable solution 👍 
I'm not going to continue now until tomorrow.